File: frontend/src/types/posting.ts
export type PostingSort = 'deadline' | 'collected_at'

export interface PostingSummary {
  id: number
  company: string
  title: string | null
  job_type: string
  industry: string
  location: string | null
  deadline: string | null
  has_cover_letter: boolean
  linkareer_url: string
  apply_url: string | null
  collected_at: string | null
}

export interface SubmissionRequirement {
  id: number
  posting_id: number
  type: string
  is_required: boolean
  detail: string | null
}

export interface CoverLetterQuestion {
  id: number
  posting_id: number
  role_name: string
  question_text: string
  char_limit: number | null
  question_order: number
}

export interface PostingDetail extends PostingSummary {
  raw_text: string
  is_image_based: boolean
  image_urls: string[]
  submission_requirements: SubmissionRequirement[]
  cover_letter_questions: CoverLetterQuestion[]
}

File: backend/app/schemas/posting.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class PostingSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    company: str
    title: str | None
    job_type: str
    industry: str
    location: str | None
    deadline: date | None
    has_cover_letter: bool
    linkareer_url: str
    apply_url: str | None
    collected_at: datetime | None


class SubmissionRequirementResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    posting_id: int
    type: str
    is_required: bool
    detail: str | None


class CoverLetterQuestionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    posting_id: int
    role_name: str
    question_text: str
    char_limit: int | None
    question_order: int


class PostingDetail(PostingSummary):
    raw_text: str
    is_image_based: bool
    image_urls: list[str]
    submission_requirements: list[SubmissionRequirementResponse]
    cover_letter_questions: list[CoverLetterQuestionResponse]